docs/cudf/source/cudf_pandas/faq.md

```diff
@@ -181,6 +181,21 @@ There are a few known limitations that you should be aware of:
    ```
 - `cudf.pandas` (and cuDF in general) is only compatible with pandas 2. Version
   24.02 of cudf was the last to support pandas 1.5.x.
+- In order for `cudf.pandas` to produce a proxy array that ducktypes as a `np.ndarray`, we actually have to wrap a valid `np.ndarray` and cannot keep the data on device with a `cupy` array. This approach incurs the overhead of an initial device-to-host (DtoH) transfer when creating a proxy array. For example,
+
+  ```python
+  import pandas as pd
+  import numpy as np
+
+  arr = pd.DataFrame("a":range(10)).values # implicit DtoH transfer
+  isinstance(arr, np.ndarrray) # returns True
+  ```
+  The reason why we do the data transfer from device to host is to ensure that the [data buffer](https://numpy.org/doc/stable/dev/internals.html#internal-organization-of-numpy-arrays) is set correctly. With the data buffer set, we can utilize other functions which require a valid data buffer.
+
+  ```python
+  import torch
+  x = torch.from_numpy(arr)
+  ```
```

Comment:
This isn't quite the right phrasing, especially for consistency with above. Above we say that we do this to have proxies ducktype as numpy arrays. I think a more accurate way to frame the whole thing would be for line 184 to say roughly "in order for cudf.pandas to produce a proxy array that ducktypes as a numpy array, we create a proxy type that actually subclasses numpy.ndarray" and remove line 190 showing the transfer. Then, here we can say "because the proxy type ducktypes as a numpy array, there are many situations where numpy will attempt to access internal members like the data buffer that make assumptions about the data being on the host or use the C API, and since we cannot proxy all of those we have to do an eager DtoH copy" and then show that in the second example. Then you can say that "with this approach we also get compatibility with third party libraries like torch". The last bit we can either add now, or after we've done the more thorough C API testing that we're discussing.

Reply:
Overall, this is clearer. This part: "that make assumptions about the data being on the host or use the C API" is not as clear to me. I think you mean the buffer assumes the data is in host memory, but it doesn't assume you're using the C API? I thinks its numpy functions that assume they can access the buffer directly. That's the assumption that does not work with our proxy arrays.

Reply:
Yeah that wasn't clear phrasing. I meant that numpy could either access the data buffer, which assumes the data is on the host, or attempt to use a C API call on a numpy array (either the buffer or otherwise). In the second case there could also be an assumption that data is on the host, or there could be an assumption that some other C member is accessible. Either way, I think what you have now is plenty clear, thanks!